### DISCUSSION TITLE
Disconnect event takes considerable time

### DISCUSSION THREAD

**Discussion by colingoodman** [miguelgrinberg/python-socketio#754] [Q&A] [answered]
I am honestly uncertain on where this ought to properly go, however I believe the [latest socketio release](https://github.com/socketio/socket.io/releases/tag/4.1.2) may be related.

**Describe the bug**
It is taking considerable time (60 seconds) for the "disconnect" event to be hit after a client breaks their connection. This was practically instantaneous as of a few days ago.

**To Reproduce**
1. Client socket connection is created.
2. Client breaks the connection.

**Expected behavior**
The "disconnect" event provided by socketio should be triggered as soon as the socket connection is broken.

**Logs**
This may or may not be related, but this message has been appearing in the flask logs on startup:

> The WebSocket transport is not available, you must install a WebSocket server that is compatible with your async mode to enable it. See the documentation for details. (further occurrences of this error will be logged with level INFO)

Googling around, previous issues seem to indicate that we need to install gevent or eventlet, but we already have these packages installed. A similar message in the logs points us to something called simple-webserver, but we (also) have this installed.

**Additional context**
We had been using the "disconnect" event provided by socketio to detect when a socket was broken by a client. We tested a version of our code from a week ago in which everything was fine, and the current problem persisted. We presume that an underlying dependency somewhere may have changed something.

It is highly probable that this is a bug caused by something we are doing; any advice or potential fixes would be immensely appreciated.

**Comment by miguelgrinberg** [comment, marked as answer]
When in doubt, please write your question in a discussion. The issues board is for actual issues or enhancements, not to troubleshoot problems. Thanks.

You are using long-polling, which by its own nature cannot detect disconnections immediately, the server has to wait some time to declare a client gone. WebSocket disconnections are detected immediately. This has always been the case there are no new changes either in the Python or JS sides that changed this behavior.

They key is given to you in the message. You are not using a WebSocket compatible web server.

> Googling around, previous issues seem to indicate that we need to install gevent or eventlet, but we already have these packages installed. A similar message in the logs points us to something called simple-webserver, but we (also) have this installed.

Installing things without really knowing what you are installing is a really bad idea. For example, only one of eventlet, gevent or simple-websocket are needed, you have to decide which one you want to use. Installing the three of them does not achieve anything, it just confuses things.

Which WebSocket package do you want to use? Keep that one, uninstall the other two, then look in the documentation for instructions on how to start your chosen WebSocket server: https://python-socketio.readthedocs.io/en/latest/server.html#deployment-strategies

**Reply by colingoodman** [reply]
I wasn't aware Github had discussions pages; I apologize for the misdirection and appreciate your patience. With this feedback we were able to get things fixed. Thanks!
